src/test/java/org/apache/commons/codec/binary/Base16Test.java

```diff
@@ -0,0 +1,634 @@
+        for (int i = 0; i < data.length; i++) {
+            buf.append(data[i]);
+            if (i != data.length - 1) {
+                buf.append(",");
+            }
+        }
+        return buf.toString();
+    }
+
+    /**
+     * Test for CODEC-265: Encode a ~1GiB file.
+     *
+     * @see <a href="https://issues.apache.org/jira/projects/CODEC/issues/CODEC-265">CODEC-265</a>
+     */
+    @Test
+    public void testCodec265_over() {
+        // almost 1GiB file to encode: 2^29 bytes
+        final int size1GiB = 1 << 29;
+
+        // Expecting a size of 2 output bytes per 1 input byte
+        final int blocks = size1GiB;
+        final int expectedLength = 2 * blocks;
+
+        // This test is memory hungry. Check we can run it.
+        final long presumableFreeMemory = BaseNCodecTest.getPresumableFreeMemory();
+
+        // Estimate the maximum memory required:
+        // 1GiB + 1GiB + ~2GiB + ~1.33GiB + 32 KiB  = ~5.33GiB
+        //
+        // 1GiB: Input buffer to encode
+        // 1GiB: Existing working buffer (due to doubling of default buffer size of 8192)
+        // ~2GiB: New working buffer to allocate (due to doubling)
+        // ~1.33GiB: Expected output size (since the working buffer is copied at the end)
+        // 32KiB: Some head room
+        final long estimatedMemory = (long) size1GiB * 4 + expectedLength + 32 * 1024;
+        Assume.assumeTrue("Not enough free memory for the test", presumableFreeMemory > estimatedMemory);
+
+        final byte[] bytes = new byte[size1GiB];
+        final byte[] encoded = new Base16().encode(bytes);
+        assertEquals(expectedLength, encoded.length);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void checkEncodeLengthBounds() {
+        final Base16 base16 = new Base16();
+        base16.encode(new byte[10], 0, 1 << 30);
+    }
+
+    @Test
+    public void testIsInAlphabet() {
+        // invalid bounds
+        Base16 b16 = new Base16(true);
+        assertFalse(b16.isInAlphabet((byte)0));
+        assertFalse(b16.isInAlphabet((byte)1));
+        assertFalse(b16.isInAlphabet((byte)-1));
+        assertFalse(b16.isInAlphabet((byte)-15));
+        assertFalse(b16.isInAlphabet((byte)-16));
+        assertFalse(b16.isInAlphabet((byte)128));
+        assertFalse(b16.isInAlphabet((byte)255));
+
+        // lower-case
+        b16 = new Base16(true);
+        for (char c = '0'; c <= '9'; c++) {
+            assertTrue(b16.isInAlphabet((byte) c));
+        }
+        for (char c = 'a'; c <= 'f'; c++) {
+            assertTrue(b16.isInAlphabet((byte) c));
+        }
+        for (char c = 'A'; c <= 'F'; c++) {
+            assertFalse(b16.isInAlphabet((byte) c));
+        }
+        assertFalse(b16.isInAlphabet((byte) ('0' - 1)));
+        assertFalse(b16.isInAlphabet((byte) ('9' + 1)));
+        assertFalse(b16.isInAlphabet((byte) ('a' - 1)));
+        assertFalse(b16.isInAlphabet((byte) ('z' + 1)));
```

Comment:
Just spotted this one. It should be `'f' + 1`

Reply:
My intention was for `'z' + 1`... I will add f+1 too